library/classes/Document.class.php

```diff
@@ -264,7 +264,7 @@ public function get_categories()
         . "WHERE `ctd`.`document_id` = ? ";
         $resultSet = sqlStatement($categories, [$this->get_id()]);
         $categories = [];
-        while ($category = sqlGetAssoc($resultSet)) {
+        while ($category = sqlFetchArray($resultSet)) {
```

Comment:
@adunsulag ,
Note this may impact your code that checks for access. For some reason the sqlGetAssoc call was always empty, which means that the  can_access call was always true no  matter what (since that function returns true when categories is empty). This change fixes the can_access call to work since the categories is now populated.